mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -1012,11 +1017,40 @@ private void CleanupFiles (string dest, HashSet<string> goodfiles)
 						continue;
 					}
 
-					if (string.IsNullOrWhiteSpace (PreserveTag)) { // only do this if there was no -preserve
+					Action actuallyDelete = () => {
 						string newname = typefile.FullName + ".remove";
-						try { System.IO.File.Delete(newname); } catch (Exception) { }
-						try { typefile.MoveTo(newname); } catch (Exception) { }
-						Console.WriteLine("Class no longer present; file renamed: " + Path.Combine(nsdir.Name, typefile.Name));
+						try { System.IO.File.Delete (newname); } catch (Exception ex) { Warning ("Unable to delete existing file: {0}", newname); }
+						try { typefile.MoveTo (newname); } catch (Exception ex) { Warning ("Unable to rename to: {0}", newname); }
+						Console.WriteLine ("Class no longer present; file renamed: " + Path.Combine (nsdir.Name, typefile.Name));
+					};
+
+					if (string.IsNullOrWhiteSpace (PreserveTag)) { // only do this if there was not a -preserve
+						using (TextWriter writer = OpenWrite (typefile.FullName, FileMode.Truncate))
```

Comment:
What is this _for_? I don't see a delete for this, meaning it wasn't moved from anywhere, but I don't understand why we're creating the file `typefile.FullName`, which would appear in the *current working directory* (no?). Is this a leftover debugging aid?